src/utils/emailAndIdExtract.ts

```diff
@@ -0,0 +1,25 @@
+import { UnprocessedResultsFromCRM } from '../types'
+
+type MassMailObject = {
+  email: string
+  id: string | number
+}
+
+export default function emailAndIdExtract (results: UnprocessedResultsFromCRM[]) {
+    const propertyObjects = results.slice(1)
+
+    const emailsAndIds = propertyObjects.reduce((resultsArray: MassMailObject[], property: UnprocessedResultsFromCRM) => {
+        if (typeof property.owner_details[0].Email === 'string') {
+            resultsArray.push(Object.assign({},
```

Comment:
is object.assign needed here?

Reply:
I thought it would be b/c I need to create a new object every time an email is found.

Reply:
``` {
                    email: property.owner_details[0].Email,
                    id: property.owner_details[0].id
                }``` creates a new object already. You'd use Object.assign if there was an existing object that you wanted to clone.

Reply:
Okay, that makes sense!